searx/engines/yahoo_news.py

```diff
@@ -22,90 +40,77 @@
-                                      lang=language)
-
-    # TODO required?
-    params['cookies']['sB'] = '"v=1&vm=p&fl=1&vl=lang_{lang}&sh=1&pn=10&rw=new'\
-        .format(lang=language)
+    params['url'] = search_url.format(
+        offset = offset,
+        query = urlencode({'p': query})
+    )
+    logger.debug("query_url --> %s", params['url'])
     return params
 
-
-def sanitize_url(url):
-    if ".yahoo.com/" in url:
-        return re.sub("\\;\\_ylt\\=.+$", "", url)
-    else:
-        return url
-
-
-# get response from search-request
 def response(resp):
     results = []
-
     dom = html.fromstring(resp.text)
 
     # parse results
-    for result in dom.xpath(results_xpath):
-        urls = result.xpath(url_xpath)
-        if len(urls) != 1:
+    for result in eval_xpath_list(dom, '//ol[contains(@class,"searchCenterMiddle")]//li'):
+
+        url = eval_xpath_getindex(result, './/h4/a/@href', 0, None)
+        if url is None:
             continue
-        url = sanitize_url(parse_url(extract_url(urls, search_url)))
-        title = extract_text(result.xpath(title_xpath)[0])
-        content = extract_text(result.xpath(content_xpath)[0])
-
-        # parse publishedDate
-        publishedDate = extract_text(result.xpath(publishedDate_xpath)[0])
-
-        # still useful ?
-        if re.match("^[0-9]+ minute(s|) ago$", publishedDate):
-            publishedDate = datetime.now() - timedelta(minutes=int(re.match(r'\d+', publishedDate).group()))
-        elif re.match("^[0-9]+ days? ago$", publishedDate):
-            publishedDate = datetime.now() - timedelta(days=int(re.match(r'\d+', publishedDate).group()))
-        elif re.match("^[0-9]+ hour(s|), [0-9]+ minute(s|) ago$", publishedDate):
-            timeNumbers = re.findall(r'\d+', publishedDate)
-            publishedDate = datetime.now()\
-                - timedelta(hours=int(timeNumbers[0]))\
-                - timedelta(minutes=int(timeNumbers[1]))
-        else:
-            try:
-                publishedDate = parser.parse(publishedDate)
-            except:
-                publishedDate = datetime.now()
-
-        if publishedDate.year == 1900:
-            publishedDate = publishedDate.replace(year=datetime.now().year)
-
-        # append result
-        results.append({'url': url,
-                        'title': title,
-                        'content': content,
-                        'publishedDate': publishedDate})
-
-    # return results
+        url = parse_url(url)
+        title = extract_text(result.xpath('.//h4/a'))
+        content = extract_text(result.xpath('.//p'))
+        img_src = eval_xpath_getindex(result, './/img/@data-src', 0, None)
+
+        item = {
+            'url': url,
+            'title': title,
+            'content': content,
+            'img_src' : img_src
+        }
+
+        pub_date = extract_text(result.xpath('.//span[contains(@class,"s-time")]'))
+        if RE_ago.search(pub_date):
```

Comment:
I don't know perl, but I'm not sure to understand the `RE` class.

The code here seems to be identical to:
```python
ago = regex_ago.search(pub_date)
if ago:
	ago = int(ago.group(0))
	if ago.group(1) == 'minute':
		pub_date = datetime.now() - timedelta(minutes=c)
       ....
```

Even simpler
```python
AGO_TIMEDELTA = {
  'minute': timedelta(minutes=1),
  'hour': timedelta(hour=1),
  'day': timedelta(day=1),
  'month': timedelta(month=1),
  'year': timedelta(year=1),
}

...
raw_pub_date= regex_ago.search(pub_date)
if ago:
	number = int(raw_pub_date.group(0))
	delta = AGO_TIMEDELTA[raw_pub_date.group(1)]
        pub_date = datetime.now() - delta * number
else:
        try:
            pub_date = parser.parse(raw_pub_date)
        except parser.ParserError:
            pub_date = None
```